Src/BootCamp.Chapter/Shop.cs

```diff
@@ -31,6 +27,13 @@ public Item[] GetItems()
         /// </summary>
         public void Add(Item item)
         {
+            Item item1 = _inventory.GetItem(item?.Name);
+            bool isSameItem = item1?.IsTheSameItem(item) ?? false;
```

Comment:
I read this as if item1 ==  null then isSameItem = false; do i read this correct? and else it checks if its the same item and returns true or false from that.

Reply:
It can be even more simple: ``item1?.IsTheSameItem(item) == true``. By doing == you ask to convert nullable<bool> into bool. If it's null, it will be false.

What you wrote is also valid, but a tiny bit harder to read. It reads: try to check if item1 is matching item. It it doesn't, return false, if it does- compare.